### PR TITLE
Added a field AwsIAMType for database user

### PR DESCRIPTION
**Pull request by coderGo93**
## Description

- Added a field for database user called `awsIAMType` in resource and datasource(s) and its test
- Convert to url query escape for `username` when it uses `GET` method to avoid special characters issues.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the MongoDB CLA
- [ ] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
Client mongo needs to be updated(merged) before this PR so I can update the vendor.

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_database_user.go [hunk 1]
```diff
@@ -139,6 +145,10 @@ func resourceMongoDBAtlasDatabaseUserRead(d *schema.ResourceData, meta interface
 		return fmt.Errorf("error setting `x509_type` for database user (%s): %s", d.Id(), err)
 	}
 
+	if err := d.Set("aws_iam_type", dbUser.AWSIAMType); err != nil {
+		return fmt.Errorf("error setting `x509_type` for database user (%s): %s", d.Id(), err)
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_database_user.go]: Error message is for the wrong type.

File: mongodbatlas/resource_mongodbatlas_database_user.go [hunk 2]
```diff
@@ -115,8 +120,9 @@ func resourceMongoDBAtlasDatabaseUserRead(d *schema.ResourceData, meta interface
 			authDatabaseName = d.Get("auth_database_name").(string)
 		}
 	}
+	dbUsername := url.QueryEscape(username)
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_database_user.go]: Perhaps a more informative name?  username_escaped or what ever best fits standard go naming?

Also I'm a bit worried about doing this across the board as we've been able to GET and DELETE with special characters in db names before (like a dash).  Let's make sure this doesn't cause any side impacts by being sure we test that thoroughly please.

File: mongodbatlas/resource_mongodbatlas_database_user_test.go [hunk 3]
```diff
@@ -71,7 +73,34 @@ func TestAccResourceMongoDBAtlasDatabaseUser_withX509Type(t *testing.T) {
 					testAccCheckMongoDBAtlasDatabaseUserAttributes(&dbUser, username),
 					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
 					resource.TestCheckResourceAttr(resourceName, "username", username),
-					resource.TestCheckResourceAttr(resourceName, "x509_type", "MANAGED"),
+					resource.TestCheckResourceAttr(resourceName, "x509_type", "CUSTOMER"),
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_database_user_test.go]: I would not remove the MANAGED test but add a 2nd test that allows for the username type for CUSTOMER x509 types.  MANAGED is far more common so we should keep the test.

File: mongodbatlas/resource_mongodbatlas_database_user_test.go [hunk 4]
```diff
@@ -342,7 +375,7 @@ func testAccMongoDBAtlasDatabaseUserWithX509TypeConfig(projectID, roleName, user
 	return fmt.Sprintf(`
 		resource "mongodbatlas_database_user" "test" {
 			username           = "%[3]s"
-			x509_type          = "MANAGED"
+			x509_type          = "CUSTOMER"
```
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_database_user_test.go]: Same as above, keep both.

File: website/docs/d/database_user.html.markdown [hunk 5]
```diff
@@ -62,6 +62,7 @@ In addition to all arguments above, the following attributes are exported:
 * `id` - The database user's name.
 * `roles` - List of user’s roles and the databases / collections on which the roles apply. A role allows the user to perform particular actions on the specified database. A role on the admin database can include privileges that apply to the other databases as well. See [Roles](#roles) below for more details.
 * `x509_type` - X.509 method by which the provided username is authenticated.
+* `aws_iam_type` - The new database user authenticates with AWS IAM credentials.
```
Comment (themantissa) [inline website/docs/d/database_user.html.markdown]: Edit - include what value may be here:
aws_iam_type - If set the database user authenticates with AWS IAM credentials.  Default is NONE, `USER` means user has AWS IAM user credentials, `ROLE` -  means user has credentials associated with an AWS IAM role.

File: website/docs/d/database_users.html.markdown [hunk 6]
```diff
@@ -68,6 +68,7 @@ In addition to all arguments above, the following attributes are exported:
 * `auth_database_name` - The user’s authentication database. A user must provide both a username and authentication database to log into MongoDB. In Atlas deployments of MongoDB, the authentication database is always the admin database.
 
 * `x509_type` - X.509 method by which the provided username is authenticated.
+* `aws_iam_type` - The new database user authenticates with AWS IAM credentials.
```
Comment (themantissa) [inline website/docs/d/database_users.html.markdown]: Same:
aws_iam_type - If set the database user authenticates with AWS IAM credentials.  Default is `NONE`, `USER` means user has AWS IAM user credentials, `ROLE` -  means user has credentials associated with an AWS IAM role.

**Review by themantissa — CHANGES_REQUESTED** [review]
A few edits and request to not remove test but add.

File: mongodbatlas/resource_mongodbatlas_database_user.go (hunk 1 above)
Comment (coderGo93) [reply mongodbatlas/resource_mongodbatlas_database_user.go]: You're right, thank you so much

File: mongodbatlas/resource_mongodbatlas_database_user.go (hunk 2 above)
Comment (coderGo93) [reply mongodbatlas/resource_mongodbatlas_database_user.go]: Sure no problem, I have tested many times with testacc and all OK when creating, getting and deleting but will do more testing to be sure of course

File: mongodbatlas/resource_mongodbatlas_database_user_test.go (hunk 3 above)
Comment (coderGo93) [reply mongodbatlas/resource_mongodbatlas_database_user_test.go]: No problem, thank you melissa

File: website/docs/d/database_user.html.markdown (hunk 5 above)
Comment (coderGo93) [reply website/docs/d/database_user.html.markdown]: I'll add this part, thank you melissa

**Review by themantissa — APPROVED** [review]
Think you addressed all the comments! Thank you - LGTM

File: mongodbatlas/resource_mongodbatlas_database_user.go [hunk 7]
```diff
@@ -53,6 +53,11 @@ func resourceMongoDBAtlasDatabaseUser() *schema.Resource {
 				Optional: true,
 				Default:  "NONE",
 			},
+			"aws_iam_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NONE",
+			},
```
Comment (PacoDw) [inline mongodbatlas/resource_mongodbatlas_database_user.go]:
```suggestion
			"aws_iam_type": {
				Type:     schema.TypeString,
				Optional: true,
				Default:  "NONE",
				ValidateFunc: validation.StringInSlice([]string{"NONE", "USER", "ROLE"}, false),
			},
```
if only these attribute must to be set in this attribute, you need to consider put this validation to ensure that the user write at least one of them.

**Review by PacoDw — CHANGES_REQUESTED** [review]

File: mongodbatlas/resource_mongodbatlas_database_user.go (hunk 7 above)
Comment (coderGo93) [reply mongodbatlas/resource_mongodbatlas_database_user.go]: Thanks for the suggestion, paco, I added it and for x509 too

**Review by PacoDw — APPROVED** [review]
LGTM!

**Pull request merged**